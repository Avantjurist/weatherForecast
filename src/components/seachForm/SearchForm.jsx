import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { itemsFetchData } from '../../actions/index';
import * as styles from './SearchForm.css';

class SeachForm extends Component {

    constructor() {
        super();
        this.state = {
            value: ''
        }
    }

    handleChange = (event) => {
        this.setState({
            value: event.target.value
        })
    };

    handleClick = () => {
        const { fetchCitiesData , history } = this.props;
        const { value } = this.state;

        history.push("/search/" + value);
        fetchCitiesData(value)
    };

    render() {
        return (
            <form className={styles.searchForm} onSubmit={e => e.preventDefault()}>
                <input className={styles.searchInput} type="text" placeholder="Your city name" value={this.state.value}
                       onChange={this.handleChange}/>
                <button className={styles.searchButton} onClick={this.handleClick}>Search</button>
            </form>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchCitiesData: (cityName) => dispatch(itemsFetchData(cityName))
    }
};

export default connect(null, mapDispatchToProps)(withRouter(SeachForm));

