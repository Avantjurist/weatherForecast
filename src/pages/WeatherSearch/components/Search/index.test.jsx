import React from 'react';
import { shallow } from "enzyme";
import SearchForm from "./index.component";

describe("Seach form", () => {

    it("render correctly SearchForm", () => {
        const output = shallow(
            < SearchForm />
        );
        expect(output).toMatchSnapshot();
    })

    it("change value on typing", () => {
        const test = "test";
        const output = shallow(
            < SearchForm />
        );

        output.find('input').simulate('change', {
            target: {
                value: test
            }
        })

        expect(output.state().value).toEqual(test)
    })

    it("call actions after click search", () => {
        const mockFetchCitiesData = jest.fn();
        const props = { history: [], fetchCitiesData: mockFetchCitiesData };
        const output = shallow(
            < SearchForm {...props} />
        )

        output.find('button').simulate('click', {})

        expect(mockFetchCitiesData).toHaveBeenCalledTimes(1);
    })

    it("nothing happen on submit form", () => {
        const output = shallow(
            < SearchForm />
        )
        const mockPreventDefault = jest.fn();

        output.find('form').simulate('submit', {
            preventDefault: mockPreventDefault
        })

        expect(output.state().value).toEqual('');
    })
})