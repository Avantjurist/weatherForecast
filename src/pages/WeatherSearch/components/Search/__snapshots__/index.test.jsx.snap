// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seach form render correctly SearchForm 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className={undefined}
          onChange={[Function]}
          placeholder="Your city name"
          type="text"
          value=""
        />,
        <button
          className={undefined}
          onClick={[Function]}
        >
          Search
        </button>,
      ],
      "className": undefined,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "onChange": [Function],
          "placeholder": "Your city name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className={undefined}
            onChange={[Function]}
            placeholder="Your city name"
            type="text"
            value=""
          />,
          <button
            className={undefined}
            onClick={[Function]}
          >
            Search
          </button>,
        ],
        "className": undefined,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "onChange": [Function],
            "placeholder": "Your city name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
